examples/cms/src/FieldMapping.tsx

```diff
@@ -0,0 +1,233 @@
+                )
+
+                setIgnoredFieldIds(new Set(ignoredFields.map(field => field.id)))
+            })
+            .catch(error => {
+                console.error(error)
+                framer.notify("Failed to load existing fields.", { variant: "error" })
+            })
+            .finally(() => {
+                setStatus("mapping-fields")
+            })
+    }, [collection, sourceFields, initialSlugFieldId])
+
+    const handleFieldNameChange = (fieldId: string, name: string) => {
+        setFields(previousFields => previousFields.map(field => (field.id === fieldId ? { ...field, name } : field)))
+    }
+
+    const handleFieldToggle = (fieldId: string) => {
+        setIgnoredFieldIds(previousIgnoredFieldIds => {
+            const updatedIgnoredFieldIds = new Set(previousIgnoredFieldIds)
+
+            if (updatedIgnoredFieldIds.has(fieldId)) {
+                updatedIgnoredFieldIds.delete(fieldId)
+            } else {
+                updatedIgnoredFieldIds.add(fieldId)
+            }
+
+            return updatedIgnoredFieldIds
+        })
+    }
+
+    const handleSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
+        event.preventDefault()
+
+        if (!selectedSlugField) {
+            // This can't happen because the form will not submit if no slug field is selected
+            // but TypeScript can't infer that.
+            console.error("There is no slug field selected. Sync will not be performed")
+            return
+        }
+
+        try {
+            setStatus("syncing-collection")
+
+            await syncCollection(
+                collection,
+                dataSource,
+                fields.filter(field => !ignoredFieldIds.has(field.id)),
+                selectedSlugField
+            )
+            await framer.closePlugin(`Synchronization successful`, {
+                variant: "success",
+            })
+        } catch (error) {
+            console.error(error)
+            framer.notify(`Failed to sync collection ${dataSource.id}. Check the logs for more details.`, {
+                variant: "error",
+            })
+        } finally {
+            setStatus("mapping-fields")
+        }
+    }
+
+    if (isLoadingFields) {
+        return <Spinner />
+    }
+
+    return (
+        <main className="mapping">
+            <hr className="sticky-top" />
+            <form onSubmit={handleSubmit}>
+                <label className="slug-field" htmlFor="slugField">
+                    Slug Field
+                    <select
+                        name="slugField"
+                        className="field-input"
+                        value={selectedSlugField ? selectedSlugField.id : ""}
+                        onChange={event =>
+                            setSelectedSlugField(
+                                possibleSlugFields.find(field => field.id === event.target.value) ?? null
+                            )
+                        }
```

Comment:
This can be simplified by doing only one operation per line:

```ts
const selectedFieldId = event.target.value
const selectedField = possibleSlugFields.find(field => field.id === selectedFieldId)
if (!selectedField) return
setSelectedSlugField(selectedField)
```